hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/result/DefaultErrorHandler.java

```diff
@@ -63,7 +63,19 @@ private void updateMeters(Subscription subscription) {
     @Override
     public void handleFailed(Message message, Subscription subscription, MessageSendingResult result) {
         hermesMetrics.meter(Meters.CONSUMER_FAILED_METER, subscription.getTopicName(), subscription.getName()).mark();
-
+        registerFailureMetrics(subscription, result);
         trackers.get(subscription).logFailed(toMessageMetadata(message, subscription), result.getRootCause());
     }
+
+    private void registerFailureMetrics(Subscription subscription, MessageSendingResult result) {
+        if (result.getStatusCode() != 0) {
```

Comment:
I think we should not dig into MessageSendingResult internals but maybe expose some methods `hasStatusCode()`, `isTimeout()`